server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -258,22 +365,78 @@ public void setName(String name) {
         this.name = name;
     }
 
+    /**
+     * Retrieves the owners with which this product is associated. If this product is not associated
+     * with any owners, this method may return an empty set, or null.
+     * <p/>
+     * Note that changes made to the set returned by this method will be reflected by this object
+     * and its backing data store.
+     *
+     * @return
+     *  The set of owners with which this product is associated
+     */
+    @XmlTransient
+    @Override
+    public Collection<Owner> getOwners() {
+        return this.owners;
+    }
 
     /**
-     * @return The product's owner/organization
+     * Associates this product with the specified owner. If the given owner is already associated
+     * with this product, the request is silently ignored.
+     *
+     * @param owner
+     *  An owner to be associated with this product
+     *
+     * @return
+     *  True if this product was successfully associated with the given owner; false otherwise
      */
-    public Owner getOwner() {
-        return this.owner;
+    public boolean addOwner(Owner owner) {
+        if (owner != null) {
+            if (this.owners == null) {
+                this.owners = new HashSet<Owner>();
+            }
+
+            return this.owners.add(owner);
+        }
+
+        return false;
     }
 
     /**
-     * Sets the product's owner.
+     * Disassociates this product with the specified owner. If the given owner is not associated
+     * with this product, the request is silently ignored.
      *
      * @param owner
-     * The new owner/organization for this product.
+     *  The owner to disassociate from this product
+     *
+     * @return
+     *  True if the product was disassociated successfully; false otherwise
+     */
+    public boolean removeOwner(Owner owner) {
+        return (this.owners != null && owner != null) ? this.owners.remove(owner) : false;
+    }
+
+    /**
+     * Sets the owners with which this product is associated.
+     *
+     * @param owners
+     *  A collection of owners to be associated with this product
+     *
+     * @return
+     *  A reference to this product
      */
-    public void setOwner(Owner owner) {
-        this.owner = owner;
+    public Product setOwners(Collection<Owner> owners) {
+        if (this.owners == null) {
+            this.owners = new HashSet<Owner>();
+        }
+
+        this.owners.clear();
+        if (owners != null) {
```

Comment:
Is this dealing with a field named `owners` and a parameter named `owners`?  That's kind of confusing if so.  How about changing the parameter name to "otherOwners" or something.